classes/user/maps/Schema.php

```diff
@@ -272,4 +281,49 @@ protected function mapByProperties(array $props, User $user, array $auxiliaryDat
 
         return $output;
     }
+
+    /**
+     * Decide if the current user can "log in as" the target user
+     */
+    protected function getPropertyCanLoginAs(User $targetUser): bool
+    {
+        $request = Application::get()->getRequest();
+        $currentUser = $request->getUser();
+        if (!$currentUser) {
+            return false; // Not logged in
+        }
+
+        // Prevent logging in as self
+        if ($currentUser->getId() === $targetUser->getId()) {
+            return false;
+        }
+
+        // This calls Validation::canUserLoginAs(...) if you have it
+        return Validation::canUserLoginAs($targetUser->getId(), $currentUser->getId());
+    }
+
+    /**
+     * Determine if the current user can merge the target user
+     */
+    protected function getPropertyCanMergeUsers(User $targetUser): bool
+    {
+        $request = Application::get()->getRequest();
+        $currentUser = $request->getUser();
+
+        // ensure a user is logged in
+        if (!$currentUser) {
+            return false;
+        }
+
+        // prevent merging oneself
+        if ($currentUser->getId() === $targetUser->getId()) {
+            return false;
+        }
+
+        // ensure user administration permissions
+        $canAdminister = Validation::getAdministrationLevel($targetUser->getId(), $currentUser->getId()) === Validation::ADMINISTRATION_FULL;
+
+        // allow merging if the current user is a site admin or has full admin rights over the target user
+        return Validation::isSiteAdmin() || $canAdminister;
```

Comment:
Should site admin role be already covered by the `Validation::getAdministrationLevel()` check?